dima/react-movies/src/components/searchBar/searchBar.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from "react";
+import SearchFilter from "./searchFilter";
+import "./searchBar.css";
+import ErrorBoundary from "../errorBoundary";
+import MovieInterface from "../../interfaces/movieInterface";
+
+interface SearchBarProps {
+  searchMovies: (
+    searchBarValue: string,
+    searchedBy: string
+  ) => MovieInterface[];
+  setSearchedMovies: (searchedMovies: MovieInterface[]) => void;
+}
+
+interface SearchBarState {
+  searchBarValue: string;
+  searchedBy: string;
+}
+
+export default class SearchBar extends React.Component<
+  SearchBarProps,
+  SearchBarState
+> {
+  state: SearchBarState = {
+    searchBarValue: "",
+    searchedBy: "title",
+  };
+  onSearchBarChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const { value: searchBarValue } = e.target;
+    console.log(searchBarValue);
+    this.setState({ searchBarValue });
+  };
+  toggleSearchCategory = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ): void => {
+    const { value } = e.currentTarget;
+    console.log(value);
+    this.setState({ searchedBy: value.toLowerCase() });
+  };
+
+  onFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const { searchBarValue, searchedBy } = this.state;
+    const searchedMovies = this.props.searchMovies(searchBarValue, searchedBy);
+    this.props.setSearchedMovies(searchedMovies);
```

Comment:
looks strange. why upper component do the sorting, then gives you data and then you give it back to the parent component, why don't you just setSearchedMovies in App where sorting is happening ?